packages/daimo-api/src/network/bundleCompression.ts

```diff
@@ -90,14 +122,24 @@ export function compressBundle(op: UserOpHex, info: CompressionInfo) {
   ret.push(`0x${m[1]}`); // sig version, validUntil, keySlot
   ret.push(`0x${m[2]}`); // sig r
   ret.push(`0x${m[3]}`); // sig s
-  ret.push(`0x${m[4]}`); // authenticator challenge
+  // no authenticator challenge. reconstructed in DaimoOpInflator.
 
   // paymaster signature, if present
-  const paymasterAddr = info.inflatorPaymaster;
+  const paymasterAddr = info.opInflatorPaymaster;
   const paymasterRegex = new RegExp(`${paymasterAddr}(.*)$`, "i");
   m = paymasterRegex.exec(op.paymasterAndData);
   if (!m) throw new Error("can't compress, bad paym.:" + op.paymasterAndData);
-  ret.push(`0x${m[1]}`); // paymaster data
+  if (m[1].length > 0) throw new Error("can't compress, has paymasterData");
 
   return concatHex(ret);
 }
+
+function compressAddr(addr: Address, nameReg: NameRegistry) {
```

Comment:
one subtle realisation I had reading this: even though nameRegistry has upgrade keys we own we can't ever redirect any transfers even if we maliciously changed name registry because the user signs the uncompressed operation with the intended address -- nice property.